### PR TITLE
fix 鎖龍蛇－スカルデット

### PR DESCRIPTION
**Pull request by salix5**
fix https://github.com/Fluorohydride/ygopro/issues/2441

### CODE REVIEW

**Comment by mercury233** [comment]
Cards like バウンドリンク, 聖邪のステンドグラス may need fix too

**Comment by salix5** [comment]
Auxiliary.PlaceOnBottom()
add
```lua
Duel.ShuffleHand(p)
```
Now the opponent cannot know which cards are selected.

**Comment by mercury233** [comment]
I think we should rename this function

**Comment by salix5** [comment]
Any suggestions?

**Review by mercury233 — CHANGES_REQUESTED** [review]

File: utility.lua [hunk 1]
```diff
@@ -2984,3 +2984,30 @@ function Auxiliary.ThisCardInGraveAlreadyReset2(e)
...
 	e:Reset()
 end
+--Player p place n cards from his hand on the bottom of his Deck in any order
+function Auxiliary.PlaceOnBottom(p,n)
+	local g=Duel.GetMatchingGroup(Card.IsAbleToDeck,p,LOCATION_HAND,0,nil)
+	if g:GetCount()<n then return end
+	Duel.ShuffleHand(p)
+	Duel.BreakEffect()
+	Duel.Hint(HINT_SELECTMSG,p,HINTMSG_TODECK)
+	local sg=g:Select(p,n,n,nil)
```
Comment (mercury233) [inline utility.lua:2994]: We should support _Dark Spirit's Mastery_ by allow passing `sg` as param

Comment (mercury233) [reply utility.lua:2994]: or another function to simply Tearlaments scripts

Comment (salix5) [reply utility.lua:2994]: So the function should be something like:
```lua
function Auxiliary.PlaceGroupOnDeckBottom(p,sg)
end
```

The script should generate sg by themselves.
Is it right?

Comment (mercury233) [reply utility.lua:2994]: I think so

**Comment by salix5** [comment]
Should we merge this one?

File: utility.lua [hunk 2]
```diff
@@ -2984,3 +2984,26 @@ function Auxiliary.ThisCardInGraveAlreadyReset2(e)
...
+		end
+	end
+	if x2>0 then
+		Duel.SortDecktop(p,1-p,x2)
+		for i=1,x2 do
+			local mg=Duel.GetDecktopGroup(1-p,1)
+			Duel.MoveSequence(mg:GetFirst(),SEQ_DECKBOTTOM)
+		end
+	end
+	return #sg2
```
Comment (mercury233) [inline utility.lua:3008]: why only return sg2

**Comment by mallu11** [comment]
@salix5 why this return #og? If there are some effects that use `xxx for each card returned`, I think this return the operation group is better (both deck and extra deck).

**Comment by mercury233** [comment]
> @salix5 why this return #og? If there are some effects that use `xxx for each card returned`, I think this return the operation group is better (both deck and extra deck).

Because other similar functions such as `Duel.Destroy`, `Duel.Remove`, `Duel.SendtoGrave` return the operated amount.

**Comment by mallu11** [comment]
> > @salix5 why this return #og? If there are some effects that use `xxx for each card returned`, I think this return the operation group is better (both deck and extra deck).
> 
> Because other similar functions such as `Duel.Destroy`, `Duel.Remove`, `Duel.SendtoGrave` return the operated amount.

I see. But #og is the operated amount of Main Deck, it should include the returned card(s) in Extra Deck.

**Pull request merged**